build/build-image/Dockerfile

```diff
@@ -31,20 +31,18 @@ RUN wget -q https://dl.google.com/go/go${GO_VERSION}.linux-amd64.tar.gz && \
 
 # install gcloud + kubectl, because it's an easy way to test/dev against kubernetes.
 WORKDIR /opt
-RUN wget -q https://dl.google.com/dl/cloudsdk/release/google-cloud-sdk.zip && unzip -q google-cloud-sdk.zip && \
-    rm google-cloud-sdk.zip && \
-    /opt/google-cloud-sdk/install.sh --usage-reporting=true --path-update=true --bash-completion=true --rc-path=/root/.bashrc
 
-# update the path for both go and gcloud
-ENV PATH /usr/local/go/bin:/go/bin:/opt/google-cloud-sdk/bin:$PATH
+# credits https://cloud.google.com/sdk/docs/install#deb
+RUN echo "deb [signed-by=/usr/share/keyrings/cloud.google.gpg] http://packages.cloud.google.com/apt cloud-sdk main" | tee -a /etc/apt/sources.list.d/google-cloud-sdk.list && curl https://packages.cloud.google.com/apt/doc/apt-key.gpg | apt-key --keyring /usr/share/keyrings/cloud.google.gpg  add - && apt-get update -y && \
+    apt-get install google-cloud-cli google-cloud-cli-gke-gcloud-auth-plugin google-cloud-cli-app-engine-go -y && \
+    echo "source /usr/share/google-cloud-sdk/completion.bash.inc" >> /root/.bashrc
+
+# update the path for go
+ENV PATH /usr/local/go/bin:/go/bin:$PATH
 
 # install go tooling for development, building and testing
 RUN go install golang.org/x/tools/cmd/goimports@latest
 
-# RUN gcloud components update
```

Comment:
Do we no longer need to try to update? Looking at the docs the command would be something like:

`sudo apt-get update && sudo apt-get --only-upgrade install google-cloud-cli google-cloud-cli-gke-gcloud-auth-plugin  google-cloud-cli-app-engine-go`

Reply:
We don't need to update since `apt` installs the latest version.

Also the line where we install all the components is here:
https://github.com/googleforgames/agones/pull/2629/files#diff-686b06cb4b89c224b0df5f76fa2bbc7b5757477074e6917a21b8040a9e6857a4R37

Reply:
Before we downloaded the latest zip but still ran components update afterwords. I wasn't sure whether something similar was still necessary but it sounds like you don't think so.